spatialdata/_core/_spatialdata_ops.py

```diff
@@ -281,3 +288,165 @@ def _describe_paths(paths: list[list[Union[int, str]]]) -> str:
+            region_key = table.uns[TableModel.ATTRS_KEY][TableModel.REGION_KEY_KEY]
+            if not (
+                isinstance(region, str)
+                and region_key is None
+                or isinstance(region, list)
+                and isinstance(region_key, str)
+            ):
+                # this code should be never reached because the validate function should have raised an exception, but
+                # let's be extra safe
+                raise RuntimeError("Tables have incompatible region keys")
+            if isinstance(region, list):
+                all_region_keys.append(region_key)
+            else:
+                assert (
+                    len(all_region_keys) == 0
+                    or len(set(all_region_keys)) == 1
+                    and all_region_keys[0] == MERGED_TABLES_REGION_KEY
+                )
+                table.obs[MERGED_TABLES_REGION_KEY] = region
+                all_region_keys.append(MERGED_TABLES_REGION_KEY)
+        if len(set(all_region_keys)) != 1:
+            raise RuntimeError("Tables have incompatible region keys")
+        merged_region_key = all_region_keys[0]
+
+        all_regions = []
+        for table in tables:
+            region = table.uns[TableModel.ATTRS_KEY][TableModel.REGION_KEY]
+            if isinstance(region, str):
+                all_regions.append(region)
+            else:
+                all_regions.extend(region)
+        all_regions = list(set(all_regions))
+        merged_region = all_regions
+
+    attr = {"region": merged_region, "region_key": merged_region_key, "instance_key": merged_instance_key}
+    merged_table = AnnData.concatenate(*tables, join="outer", uns_merge="same")
+
+    # remove the MERGED_TABLES_REGION_KEY column if it has been added (the code above either adds that column
+    # to all the tables, either it doesn't add it at all)
+    for table in tables:
+        if MERGED_TABLES_REGION_KEY in table.obs:
+            del table.obs[MERGED_TABLES_REGION_KEY]
+
+    merged_table.uns[TableModel.ATTRS_KEY] = attr
+    merged_table.obs[merged_region_key] = merged_table.obs[merged_region_key].astype("category")
+    TableModel().validate(merged_table)
+    return merged_table
+
+
+##
+
+
+def concatenate(sdatas: list[SpatialData], omit_table: bool = False) -> SpatialData:
+    """Concatenate a list of spatial data objects.
+
+    Parameters
+    ----------
+    sdatas
+        The spatial data objects to concatenate.
+
+    Returns
+    -------
+    SpatialData
+        The concatenated spatial data object.
+    """
+    from spatialdata._core._spatialdata import SpatialData
+
+    assert type(sdatas) == list
+    assert len(sdatas) > 0
+    if len(sdatas) == 1:
+        return sdatas[0]
+
+    if not omit_table:
+        list_of_tables = [sdata.table for sdata in sdatas if sdata.table is not None]
+        merged_table = concatenate_tables(list_of_tables)
```

Comment:
if this is where it's used, then I would 💯 make it private func

Reply:
Done, thanks.